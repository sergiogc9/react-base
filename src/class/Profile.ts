export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
