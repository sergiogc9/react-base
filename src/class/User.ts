export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
