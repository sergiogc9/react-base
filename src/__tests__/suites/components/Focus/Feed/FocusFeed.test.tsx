import React from "react";
import { shallow } from "enzyme";

import { FocusObject } from "@src/class/Focus";
import FocusFeed from "@src/components/Focus/Feed/FocusFeed";

const focus: FocusObject = {
	deleted_at: new Date(1549287550563),
	feeds: { online: [], socialmedia: [], print: [] },
	id: "focus-id-1",
	inserted_at: new Date(1549287550563),
	name: "focus-name-1",
	updated_at: new Date(1549287550563),
	url_logo: "http://fake-url.com",
	acl_users: [
		{
			user_id: 'fake-user-id-1',
			discover_role: 'viewer',
			last_name: "Name",
			first_name: "User"
		},
		{
			user_id: 'fake-user-id-2',
			discover_role: 'viewer',
			last_name: 'User 2',
			first_name: 'Name 2'
		}
	]
};

const focus2: FocusObject = {
	...focus,
	id: "focus-id-2",
	name: "focus-name-2",
};

const focusList: FocusObject[] = [focus, focus2];

describe("FocusFeed", () => {
	let wrapper: any;
	let onFetchFocusListMock = jest.fn();
	let onResetFeedMock = jest.fn();
	let onFetchFeedMock = jest.fn();

	beforeEach(() => {
		onFetchFocusListMock = jest.fn();
		onResetFeedMock = jest.fn();
		onFetchFeedMock = jest.fn();
		wrapper = shallow(<FocusFeed
			isCreatePage={true}
			urlFocusId={focus.id}
			urlFeedType="online"
			focusList={focusList}
			feedType={null}
			loadingSaveFeed={false}
			onFetchFocusList={onFetchFocusListMock}
			onResetFeed={onResetFeedMock}
			onFetchFeed={onFetchFeedMock}
		/>);
	});

	it("Online feed reseting feed", () => {
		wrapper = shallow(<FocusFeed
			isCreatePage={false}
			urlFocusId={focus.id}
			urlFeedId="feed-id"
			focusList={null}
			feedType="online"
			loadingSaveFeed={false}
			onFetchFocusList={onFetchFocusListMock}
			onResetFeed={onResetFeedMock}
			onFetchFeed={onFetchFeedMock}
		/>);
		expect(wrapper.debug()).toMatchSnapshot();
		expect(wrapper.find('CircularProgress')).toHaveLength(1);
		expect(onFetchFeedMock).toHaveBeenCalledTimes(1);
		expect(onFetchFeedMock).toHaveBeenCalledWith(focus.id, "feed-id");
	});

	it("Online feed fetching feed", () => {
		expect(wrapper.debug()).toMatchSnapshot();
		expect(wrapper.find('CircularProgress')).toHaveLength(1);
		expect(onResetFeedMock).toHaveBeenCalledTimes(1);
	});

	it("Create page without focus list not reseting feed", () => {
		onResetFeedMock = jest.fn();
		wrapper = shallow(<FocusFeed
			isCreatePage={true}
			urlFocusId={focus.id}
			focusList={null}
			feedType="online"
			loadingSaveFeed={false}
			onFetchFocusList={onFetchFocusListMock}
			onResetFeed={onResetFeedMock}
			onFetchFeed={onFetchFeedMock}
		/>);
		expect(onResetFeedMock).toHaveBeenCalledTimes(0);
	});

	it("Create page without focus list reset feed after changing focus list props", () => {
		onResetFeedMock = jest.fn();
		wrapper = shallow(<FocusFeed
			isCreatePage={true}
			urlFocusId={focus.id}
			focusList={null}
			feedType="online"
			loadingSaveFeed={false}
			onFetchFocusList={onFetchFocusListMock}
			onResetFeed={onResetFeedMock}
			onFetchFeed={onFetchFeedMock}
		/>);
		expect(onResetFeedMock).toHaveBeenCalledTimes(0);
		wrapper.setProps({ focusList });
		expect(onResetFeedMock).toHaveBeenCalledTimes(1);
	});

	it("Create page without focus list not cal reset feed after changing focus list props", () => {
		onResetFeedMock = jest.fn();
		wrapper = shallow(<FocusFeed
			isCreatePage={true}
			urlFocusId="non-existing-focus"
			focusList={null}
			feedType="online"
			loadingSaveFeed={false}
			onFetchFocusList={onFetchFocusListMock}
			onResetFeed={onResetFeedMock}
			onFetchFeed={onFetchFeedMock}
		/>);
		expect(onResetFeedMock).toHaveBeenCalledTimes(0);
		wrapper.setProps({ focusList });
		expect(onResetFeedMock).toHaveBeenCalledTimes(0);
	});

	it("Online feed without focus list", () => {
		wrapper = shallow(<FocusFeed
			isCreatePage={true}
			urlFocusId={focus.id}
			focusList={null}
			feedType="online"
			loadingSaveFeed={false}
			onFetchFocusList={onFetchFocusListMock}
			onResetFeed={onResetFeedMock}
			onFetchFeed={onFetchFeedMock}
		/>);
		expect(wrapper.debug()).toMatchSnapshot();
		expect(wrapper.find('CircularProgress')).toHaveLength(1);
		expect(onFetchFocusListMock).toHaveBeenCalledTimes(1);
	});

	it("Online create feed component snapshot", () => {
		wrapper.setProps({ urlFeedType: "online", feedType: "online" });
		expect(wrapper.debug()).toMatchSnapshot();
		expect(onFetchFocusListMock).toHaveBeenCalledTimes(0);
	});

	it("Social create feed component snapshot", () => {
		wrapper.setProps({ urlFeedType: "socialmedia", feedType: "socialmedia" });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Print create feed component snapshot", () => {
		wrapper = shallow(<FocusFeed
			isCreatePage={true}
			urlFocusId={focus.id}
			focusList={null}
			feedType="print_percolator"
			loadingSaveFeed={false}
			onFetchFocusList={onFetchFocusListMock}
			onResetFeed={onResetFeedMock}
			onFetchFeed={onFetchFeedMock}
		/>);
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Online existing feed component snapshot", () => {
		wrapper.setProps({ urlFeedType: null, feedType: "online" });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Social existing feed component snapshot", () => {
		wrapper.setProps({ urlFeedType: null, feedType: "socialmedia" });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Print existing feed component snapshot", () => {
		wrapper.setProps({ urlFeedType: null, feedType: "print_percolator" });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component snapshot with save feed loading", () => {
		wrapper.setProps({ loadingSaveFeed: true });
		expect(wrapper.debug()).toMatchSnapshot();
	});

});
