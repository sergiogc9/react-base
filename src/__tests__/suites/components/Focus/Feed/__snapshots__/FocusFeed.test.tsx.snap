// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusFeed Component snapshot with save feed loading 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <div className=\\"focusFeedLoading\\">
    <CircularProgress id=\\"focusFeedLoading\\" scale={1.25} style={{...}} determinateDashoffset={187} centered={true} />
  </div>
</div>"
`;

exports[`FocusFeed Online create feed component snapshot 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <Connect(FocusFeedFacets) />
  <div className=\\"main-drawer\\">
    <Connect(FocusFeedForm) />
    <Connect(SearchResults) />
    <Connect(SearchPaginator) />
  </div>
</div>"
`;

exports[`FocusFeed Online existing feed component snapshot 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <Connect(FocusFeedFacets) />
  <div className=\\"main-drawer\\">
    <Connect(FocusFeedForm) />
    <Connect(SearchResults) />
    <Connect(SearchPaginator) />
  </div>
</div>"
`;

exports[`FocusFeed Online feed fetching feed 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <div className=\\"focusFeedLoading\\">
    <CircularProgress id=\\"focusFeedLoading\\" scale={1.25} style={{...}} determinateDashoffset={187} centered={true} />
  </div>
</div>"
`;

exports[`FocusFeed Online feed reseting feed 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <div className=\\"focusFeedLoading\\">
    <CircularProgress id=\\"focusFeedLoading\\" scale={1.25} style={{...}} determinateDashoffset={187} centered={true} />
  </div>
</div>"
`;

exports[`FocusFeed Online feed without focus list 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <div className=\\"focusFeedLoading\\">
    <CircularProgress id=\\"focusFeedLoading\\" scale={1.25} style={{...}} determinateDashoffset={187} centered={true} />
  </div>
</div>"
`;

exports[`FocusFeed Print create feed component snapshot 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <div className=\\"focusFeedLoading\\">
    <CircularProgress id=\\"focusFeedLoading\\" scale={1.25} style={{...}} determinateDashoffset={187} centered={true} />
  </div>
</div>"
`;

exports[`FocusFeed Print existing feed component snapshot 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <Connect(FocusFeedFacets) />
  <div className=\\"main-drawer\\">
    <Connect(FocusFeedForm) />
    <Connect(SearchResults) />
    <Connect(SearchPaginator) />
  </div>
</div>"
`;

exports[`FocusFeed Social create feed component snapshot 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <Connect(FocusFeedFacets) />
  <div className=\\"main-drawer\\">
    <Connect(FocusFeedDefinitionSocial) />
    <Connect(FocusFeedForm) />
    <Connect(SearchResults) />
    <Connect(SearchPaginator) />
  </div>
</div>"
`;

exports[`FocusFeed Social existing feed component snapshot 1`] = `
"<div id=\\"focusFeed\\">
  <Connect(FocusFeedHeader) />
  <Connect(FocusFeedFacets) />
  <div className=\\"main-drawer\\">
    <Connect(FocusFeedDefinitionSocial) />
    <Connect(FocusFeedForm) />
    <Connect(SearchResults) />
    <Connect(SearchPaginator) />
  </div>
</div>"
`;
